tools/index.ts

```diff
@@ -1,6 +1,60 @@
-// import {TS} from './ts-code';
-//console.log('>>>', TS.modules());
+import os from 'os';
+import fs from 'fs';
+import path from 'path';
+import {Code, parseApi} from './api';
+import apiJson from './api.json';
+import {TSCode} from './ts-code';
+import {Docs} from './docs';
+import program from 'commander';
 
-import {generateDocs} from './generate-docs';
-generateDocs();
+function resolveDir(out: string): string {
+    if (!out.startsWith('~/')) {
+        return out;
+    }
+    return path.resolve(os.homedir(), out.substr(2));
+}
 
+function writeText(outDir: string, fileName: string, text: string) {
+    console.log(fileName);
+    fs.writeFileSync(path.resolve(outDir, fileName), text, 'utf8');
+}
+
+function docs(options: any) {
+    const outDir = resolveDir(options.out);
+    console.log('Generate docs to:', outDir);
+    const code = new TSCode(parseApi(apiJson));
+    const docs = new Docs(code);
+    writeText(outDir, 'modules.md', docs.modules());
+    for (const module of docs.api.modules) {
+        writeText(outDir, `mod_${module.name}.md`, docs.module(module));
+    }
+}
+
+function codeFromLanguage(language: string): Code {
+    const api = parseApi(apiJson);
+    switch (language) {
+    case 'ts':
+        return new TSCode(api);
+    default:
+        console.error(`error: unsupported language ${language}`);
+        process.exit(1);
+    }
+}
+
+function binding(options: any) {
+    const language: string = options.language;
+    const code = codeFromLanguage(language);
+    const outDir = resolveDir(options.out);
+    console.log(`Generate ${language} binding to: ${outDir}`);
+    writeText(outDir, `modules.${language}`, code.modules());
```

Comment:
Is it possible to automatically format produced code with target language formater?

Reply:
It isn't possible, but we can produce well formatted code.